src/Portage/EBuild.hs

```diff
@@ -102,9 +104,7 @@ showEBuild now ebuild =
   ss ("# ebuild generated by hackport " ++ hackportVersion ebuild). nl.
   sconcat (map (\(k, v) -> ss "#hackport: " . ss k . ss ": " . ss v . nl) $ used_options ebuild).
   nl.
-  (case cabal_pn ebuild of
-     Nothing -> id
-     Just pn -> ss "CABAL_PN=". quote pn. nl).
+  beforeInherit.
```

Comment:
This is the only change I would make. It leaves out `CABAL_HACKAGE_REVISION` if it equals `"0"`. The `beforeInherit` function also prevents any unneeded empty lines that `pkgcheck` would complain about.

Reply:
I tested this on a package version with no revisions and it works correctly.